### PR TITLE
since no unique except thrown with ELF32/ELF64 violation. Examine err…

### PR DESCRIPTION
**Pull request by taartspi**
…or message for wrong ELF class. If found in except data throw detailed message to try and inform the user of the error. Change in normal and pi4j.library.path set to library location.

### CODE REVIEW

**Comment by taartspi** [comment]
Frank commented on a post where the user was told they had a mixture of 32 and 64 bit shared objects.  He asked if we could provide some explanation in the exception we throw. Since the condition does not throw a unique excep there is no clean way to know this happened. So I check the error message to see if this binary mis-match occurred.  I updated the two normal paths in both the linuxfs and pigpio loader.  Could not test these error paths in the linuxfs loader as the test case I use does run, but  linuxfs does not read the gpio.

File: libraries/pi4j-library-linuxfs/src/main/java/com/pi4j/library/linuxfs/util/NativeLibraryLoader.java [hunk 1]
```diff
@@ -124,13 +124,23 @@ else if (libpath.equalsIgnoreCase("local")) {
...
-                        "; UNDERLYING EXCEPTION: [" + ex.getClass().getName() + "]=" + ex.getMessage());
+                catch (Exception  | UnsatisfiedLinkError ex){
+                    String exceptMessage;
+                    if (ex.getMessage().contains("wrong ELF class")) {
+                        exceptMessage = "Pi4J was unable to link the native library [" +
+                            path + "] embedded  inside this JAR [" +
+                            NativeLibraryLoader.class.getProtectionDomain().getCodeSource().getLocation().getPath() +
+                            "].  The exception indicates a mismatch of architecture. armhf/ELFCLASS32 aarch64/ELFCLASS64 \n" +
+                            " UNDERLYING EXCEPTION: [" + ex.getClass().getName() + "]=" + ex.getMessage();
+                    }else{
```
Comment (eitch) [inline libraries/pi4j-library-linuxfs/src/main/java/com/pi4j/library/linuxfs/util/NativeLibraryLoader.java:135]: can you fix the formatting?

File: libraries/pi4j-library-linuxfs/src/main/java/com/pi4j/library/linuxfs/util/NativeLibraryLoader.java [hunk 2]
```diff
@@ -124,13 +124,23 @@ else if (libpath.equalsIgnoreCase("local")) {
...
                     System.load(path);
                 }
-                catch (Exception ex){
-                    //throw this error
-                    throw new UnsatisfiedLinkError("Pi4J was unable load the native library [" +
-                        libName + "] from the user defined library path.  The system property 'pi4j.library.path' is defined as [" +
-                        libpath + "]. Please make sure the defined the 'pi4j.library.path' " +
-                        "system property contains the correct absolute library path." +
-                        "; UNDERLYING EXCEPTION: [" + ex.getClass().getName() + "]=" + ex.getMessage());
+                catch (Exception  | UnsatisfiedLinkError ex){
```
Comment (eitch) [inline libraries/pi4j-library-linuxfs/src/main/java/com/pi4j/library/linuxfs/util/NativeLibraryLoader.java:127]: can you put the UnsatisfiedLinkError in its own block?

File: libraries/pi4j-library-linuxfs/src/main/java/com/pi4j/library/linuxfs/util/NativeLibraryLoader.java [hunk 3]
```diff
@@ -124,13 +124,23 @@ else if (libpath.equalsIgnoreCase("local")) {
                     // load library from user defined absolute path provided via pi4j.library.path}
                     System.load(path);
                 }
-                catch (Exception ex){
```
Comment (eitch) [inline libraries/pi4j-library-linuxfs/src/main/java/com/pi4j/library/linuxfs/util/NativeLibraryLoader.java:127]: here the formatting is also not ok

**Review by eitch — COMMENTED** [review]
I like your changes, thanks a lot. Can you please format your code, and put the UnsatisfiedLinkError in their own blocks? Since you're using an if/else, you might as well put it in their own block

**Comment by taartspi** [comment]
Did a second push/pull.   Made sure my workspace had the .editconfig changes as shown on the web.  I only reformatted the area I changed.
Broke the catches into individuals.   As I don't know what conditions might throw UnsatisfiedLinkError I did leave the search for the wrong ELF class. Believer that bad messages are worse than none, so if not certain this is and ELF width error the more generic message is rethrown.

**Review by eitch — APPROVED** [review]

**Comment by eitch** [comment]
Ok, merged it now

**Pull request merged**